lib/ret/remote_oidc_token.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Ret.RemoteOIDCToken do
+  @moduledoc """
+  This represents an OpenID Connect token returned from a remote service.
+  The public keys will be configured by an admin for a particular setup.
+  """
+  use Guardian,
+    otp_app: :ret,
+    secret_fetcher: Ret.RemoteOIDCTokenSecretsFetcher
+
+  def subject_for_token(_, _), do: {:ok, nil}
+  def resource_from_claims(_), do: {:ok, nil}
+end
+
+defmodule Ret.RemoteOIDCTokenSecretsFetcher do
+  def fetch_signing_secret(_mod, _opts) do
+    {:error, :not_implemented}
+  end
+
+  def fetch_verifying_secret(mod, token_headers, _opts) do
+    IO.inspect(token_headers)
+
+    # TODO use KID to look up the key. Do we want to bake it into the config at setup time? Fetch and cache? Should it be optional?
```

Comment:
I could not find it formally defined, but looking at how other libraries handle this they seem to either cache on an interval, or re-pulling keys if a key id that is not yet in the cache is used. My plan is to do a Cachex cache storing each key by id, with some long expiriy time (like 30 days) and then purging the cache and refetching if we see a key we don't already have. 

I got the impression this is a pretty rare event for most normal cases. If a user wanted to manually invalidate the cache they could do so by just restarting their reticulum servers, since I dont think we write Cachex caches to disk at all.